src/webgpu/api/operation/rendering/3d_texture_slices.spec.ts

```diff
@@ -0,0 +1,363 @@
+export const description = `
+Test rendering to 3d texture slices.
+- Render to same slice on different render pass, different textures, or texture [1, 1, N]'s different mip levels
+- Render to different slices at mip levels on same texture in render pass
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { kBytesPerRowAlignment } from '../../../util/texture/layout.js';
+
+const kSize = 4;
+
+class F extends GPUTest {
+  create3DTexture(
+    options: {
+      format?: GPUTextureFormat;
+      width?: number;
+      height?: number;
+      depthOrArrayLayers?: number;
+      mipLevelCount?: number;
+      usage?: GPUTextureUsageFlags;
+    } = {}
+  ): GPUTexture {
+    const {
+      format = 'rgba8unorm',
+      width = kSize,
+      height = kSize,
+      depthOrArrayLayers = 1,
+      mipLevelCount = 1,
+      usage = GPUTextureUsage.RENDER_ATTACHMENT,
+    } = options;
+
+    return this.device.createTexture({
+      size: [width, height, depthOrArrayLayers],
+      dimension: '3d',
+      format,
+      mipLevelCount,
+      usage,
+    });
+  }
```

Comment:
nit: This helper is quite verbose but saves very little space at the call sites. I would remove it and just do normal createTexture calls.